tensorflow/core/ops/nn_ops.cc

```diff
@@ -409,10 +409,12 @@ The size of 1D Tensors matches the dimension C of the 4D Tensors.
 y_backprop: A 4D Tensor for the gradient with respect to y.
 x: A 4D Tensor for input data.
 scale: A 1D Tensor for scaling factor, to scale the normalized x.
-reserve_space_1: A 1D Tensor for the computed batch mean, to be reused
+reserve_space_1: A 1D Tensor for the computed batch mean when is_training is True,
```

Comment:
How about something like this?

A 1D Tensor for the computed batch mean when is_training is True, to be reused in the gradient computation; or the population mean when is_training is False, to be used in the second-order gradient computation.

And the same for reserve_space_2.

Reply:
When is_training is False, pop_mean/pop_variance is needed for first-order gradient computation as well.

Reply:
Ah, ok, then "to be used in the first-order and second-order gradient computation."